pydicom/dataelem.py

```diff
@@ -212,8 +221,15 @@ def __init__(self,
         self.private_creator = None
 
     @classmethod
-    def from_json(cls, dataset_class, tag, vr, value, value_key,
-                  bulk_data_uri_handler=None):
+    def from_json(
+        cls: "DataElement",
+        dataset_class: "Dataset",
```

Comment:
Usually type hints refer to an instance - I'm not aware of how classes themselves are typed.  Does `cls: "DataElement"` work properly here (similarly for `dataset_class`), or should it be `Type["DataElement"]` or something similar?

Reply:
As far as I understand it, these are implicitly changed into a [ForwardRef](https://docs.python.org/3/library/typing.html#typing.ForwardRef).

Reply:
I'm not sure the ForwardRef helps;  I'm asking about the type since we're passing a class, not an instance... 

```python
>>> ds = Dataset()
>>> type(ds)
<class 'pydicom.dataset.Dataset'>
>>> type(Dataset)
<class 'type'>
```
Maybe it's all fine, and I just don't know enough about the typing processes in Python and how they are labeled.

Reply:
[See also](https://www.python.org/dev/peps/pep-0484/#forward-references)

Reply:
Here's an example where the [class is used](https://github.com/pydicom/pydicom/blob/master/pydicom/tag.py#L38) in the type hint
```python
>>> from typing import get_type_hints
>>> from pydicom.tag import Tag
>>> get_type_hints(tag.Tag)
{'arg': typing.Union[int, str, typing.Tuple[int]], 'arg2': typing.Union[int, NoneType], 'return': <class 'pydicom.tag.BaseTag'>}
>>>
```

Reply:
I'm still not sure we're talking about the same thing - those still seem to be instances.  I'm talking about something like [this discussion](https://github.com/python/typing/issues/254#issuecomment-661778929).  The `from_json` `cls` is not a DataElement (instance), it is a _class_ which is going to be called to create an instance of it's type.  As per my code in previous comment, the 'type' of `cls` is `type`.  Like I said, maybe it doesn't matter, but that last example quoting from Guido himself uses `Type`

Reply:
Ah, sorry, misunderstood. Updated

Reply:
I think `dataset_class` also needs to be a Type, since it is used to create an instance eventually.

Reply:
Done